packages/bpk-component-image/src/withLazyLoading.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { wrapDisplayName } from 'bpk-react-utils';
+
+export default function withLazyLoading(Component) {
+  class WithLazyLoading extends React.Component {
+    constructor() {
+      super();
+
+      this.setInView = this.setInView.bind(this);
+      this.removeEventListeners = this.removeEventListeners.bind(this);
+      this.checkInView = this.checkInView.bind(this);
+      this.isInViewPort = this.isInViewPort.bind(this);
+      this.supportsPassiveEvents = this.supportsPassiveEvents.bind(this);
+
+      this.state = {
+        inView: false,
+      };
+    }
+
+    componentDidMount() {
+      const passiveArgs = this.supportsPassiveEvents() ? { passive: true } : false;
+      document.addEventListener('scroll', this.checkInView, passiveArgs);
```

Comment:
Why not? `componentDidMount` gets fired only in the browser (after mounting), not on the server.